src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -29,7 +30,7 @@ public class HttpSource : IDisposable
         private readonly Func<Task<HttpHandlerResource>> _messageHandlerFactory;
         private readonly Uri _baseUri;
         private HttpClient _httpClient;
-        private int _authRetries;
+        private Dictionary<string, AmbientAuthenticationState> _authStates = new Dictionary<string, AmbientAuthenticationState>();
```

Comment:
As a possible solution we can auto-expire the state say after 1 hour. That would be similar to what we did for image download failures.